### ISSUE TITLE
Make modifying layout easier

### ISSUE THREAD

**Issue by aplavin** [ipython/ipywidgets#715]
Currently it's not very convenient to change something in a widget layout, especially if you need to change several properties, for example:

``` python
w = Button()
w.layout.width = '100px'
w.layout.height = '50px'
```

It could look much better:

``` python
w = Button().with_layout(width='100px', height='50px')
```

Personally I now use dumb monkey-patching for this:

``` python
def _with_layout(self, **kwargs):
    for k, v in kwargs.items():
        setattr(self.layout, k, v)
    return self

iw.DOMWidget.with_layout = _with_layout
```

and would really like to see something like this in `ipywidgets`!

**Comment by dmadeka** [comment]
You can always create a new `Layout` class and pass it to each widget right?

```python
layout = Layout(width='100px', height='50px')

w = Button(layout=layout)
```

**Comment by aplavin** [comment]
Well, I guess something like `w = Button(layout=Layout(width='100px', height='50px'))` looks almost good enough :) Didn't think about it, and didn't find in docs either, thanks!

**Comment by dmadeka** [comment]
Ugh, yeah, it's all the way in the bottom as part of the Box styling. I don't see it before either.

http://ipywidgets.readthedocs.io/en/latest/examples/Widget%20Styling.html

**Comment by aplavin** [comment]
The main point here was allowing to set layout "inline", at the same place where the widget is created. I think this should be the default way of setting layout traits in documentation, e.g. instead of this:

```python
s = Button(description='(50% width, 80px height) button')
s.layout.width = '50%'
s.layout.height = '80px'
s
```

use this:

```python
s = Button(description='(50% width, 80px height) button',
           layout=Layout(width = '50%', height = '80px'))
s
```

It looks way cleaner for me.

**Comment by dmadeka** [comment]
Agreed, or at least it should be featured often. Though the other way can be done after displaying the widget, which is kind of the interactive component

**Comment by jdfreder** [comment]
Agreed the docs should be updated.  The docs probably look like that because it was the simplest quick patch to transition them from the old API to the new layout API.

**Comment by gwvr** [comment]
Further to the previous comments, I've tried and failed to layout widgets to my satisfaction.
I'd really appreciate improved documentation on the usage of the layout widgets - in particular the usage of `Box`, `HBox` & `VBox` with `interact`, `@interact` & `interactive`.

The issues I've encountered are:

1. Layout of multiple widgets in a two column layout - I've only managed to achieve this with a kludgy attempt using `display` with the children of `interactive`, and I can't believe this is the best/correct way.
2. `interact` printing the function name between my widgets and plots - not pretty
3. My kludgy implementation of `display` with `interactive` doesn't display my plot when the cell is first run, only when a widget is used.

I can supply my test notebook if required.
